src/components/Construction/index.tsx

```diff
@@ -0,0 +1,275 @@
+import { RIGHT_BAR_WIDTH } from "@/constants/designPage";
+import { Box, Center, Button } from "@chakra-ui/react";
+import { useState, useRef, useEffect } from "react";
+import Konva from "konva";
+import { useDispatch } from "react-redux";
+import { Stage, Layer, Line, Rect, Text, Image } from "react-konva";
+import { useStoreSelector } from "@/store/hooks";
+import {
+  COORDINATES_AREA_WIDTH,
+  IMAGE_COVERAGE,
+  DRAW_DELAY,
+  PIXEL_RATIO,
+} from "@/constants/construction";
+import { findDistinctColor } from "@/utils/findDistinctColor";
+import { changeConstructionPdfSrc } from "@/store/reducer/constructionSlice";
+import {
+  switchConstructionOpen,
+  switchConstructionMounted,
+} from "@/store/reducer/buttonToggleSlice";
+
+interface Coordinates {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+  text: string;
+}
+
+const Construction = () => {
+  const dispatch = useDispatch();
+  const isConstructionOpen = useStoreSelector((state) => state.buttonToggle.isConstructionOpen);
+  const pdfRef = useRef<Konva.Stage>(null);
+  const imgSrc = useStoreSelector((state) => state.construction.constructionSrc);
+  const constructionInfo = useStoreSelector((state) => state.construction.constructionInfo);
+  const tiles = useStoreSelector((state) => state.tile.present.court);
+  const tilesColor = tiles.map((item) => item.color);
```

Comment:
It can be memories by using React hooks such as useCallback or useMemo to reduce the compute when rerender. Same with some other methods and variables on this file.

Reply:
Some funcs and variables has been rewritten with useCallback and useMemo.